tests/testsuite/http_registry.rs

```diff
@@ -15,7 +15,9 @@ use std::fs;
 use std::path::Path;
 
 fn cargo(p: &cargo_test_support::Project, s: &str) -> cargo_test_support::Execs {
-    p.cargo(s)
+    let mut e = p.cargo(s);
+    e.arg("-Zhttp-registry").masquerade_as_nightly_cargo();
+    e
```

Comment:
I think we also need a test of the unstable option itself. I.E. That the functionality does not work without the unstable flag, and that the unstable flag does not work on stable. I think we have these tests for the other unstable flags. If I'm wrong about that sorry.

Reply:
I'll add a test that verifies using `sparse+` registries requires the `-Z` flag.

Cargo already has a single test that verifies that `-Z` flags don't work on stable, so I don't think we need a special one here.

Reply:
sounds good, thank you.